src/DotVVM.Framework/Routing/RouteHelper.cs

```diff
@@ -35,5 +36,54 @@ public static void AutoDiscoverRoutes(this DotvvmRouteTable table, IRoutingStrat
                 table.Add(route.RouteName, route);
             }
         }
+
+        public static void AssertConfigurationIsValid(this DotvvmConfiguration config)
```

Comment:
And where is it invoked?

Reply:
You can invoke this method after registration of dotvvm by UseDotVVM that returns DotvvmConfiguration. Then you can call config.AssertConfigurationIsValid() to verify that you configuration is valid. The usage of this method is same as AutoMapper.AssertConfigurationIsValid(). You can use it if you need it but it is optional.

Reply:
In that case it's fairly useless, I guess.

Reply:
@adamjez  The purpose of this validation method is to let developer know that something in configuration is incorrectly registered. As developer, you want to know whether all routes in your app or src of markup controls are correct. This behavior is same as AutoMapper do this. So I think that **502.5 Process Failure** is correct behavior.

Of course when you want to use this somewhere else you can create a postback and there call `Context.Configuration.AssertConfigurationIsValid()`. Then it throws exception in postback and you can catch it and provide a more info at output.

Reply:
Hmm, this is something you don't want to call multiple times. I believe it should not be used in production at all. Or are there any reasons to do so?